components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java

```diff
@@ -186,6 +186,10 @@ public static DefaultClaimHandler getInstance() {
         filterSPClaims(spRequestedClaimMappings, localUnfilteredClaims, spUnfilteredClaims, spFilteredClaims,
                        localToSPClaimMappings);
 
+        if(spRequestedClaimMappings.isEmpty()) {
+            spFilteredClaims = spUnfilteredClaims;
+        }
```

Comment:
This statement is wrong. 'spFilteredClaims' means filtering and reducing the claims coming from the IDP side to the SP side requested claims.
@IsuraD : can you please confirm?